openmdao/core/component.py

```diff
@@ -615,12 +586,110 @@ def declare_partials(self, of, wrt, dependent=True,
         val : float or ndarray of float or scipy.sparse
             Value of subjacobian.  If rows and cols are not None, this will
             contain the values found at each (row, col) location in the subjac.
+        method : str
+            The type of approximation that should be used. Valid options include:
+                - 'fd': Finite Difference, 'cs': Complex Step, 'exact': use the component
```

Comment:
suggest 'compute' as more descriptive than `exact` (and semantically links to the `compute_partials` method)

Reply:
but you also 'compute' them via FD or CS.

Reply:
the framework does in those cases, versus the user implementing the `compute` method, right? so `user` maybe...  just seems like `exact` doesn't really capture the idea that the user has to implement the `compute` function...  /shrug